readthedocs/builds/models.py

```diff
@@ -298,8 +299,15 @@ def get_absolute_url(self):
         # but this is much simpler to handle since we only link them a couple places for now
         if self.type == EXTERNAL:
             # Django's static file serving doesn't automatically append index.html
-            url = f'{settings.EXTERNAL_VERSION_URL}/html/' \
-                f'{self.project.slug}/{self.slug}/index.html'
+            scheme = 'https' if settings.PUBLIC_DOMAIN_USES_HTTPS else 'http'
+            path = self.project.get_storage_path(
+                type_='html',
+                version_slug=self.slug,
+                version_type=self.type,
+                include_file=False,
+            )
+            url = f'{scheme}://{self.project.subdomain()}/' \
```

Comment:
I feel like calling functions in f-strings is a bit weird. Probably better to have it be a variable? I don't know though.